apps/api/migrations/expire-at/expire-at.migration.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  MessageRepository,
+  NotificationRepository,
+  ExecutionDetailsRepository,
+  JobRepository,
+  OrganizationRepository,
+  EnvironmentRepository,
+  JobStatusEnum,
+} from '@novu/dal';
+import { addMonths } from 'date-fns';
+
+const messageRepository = new MessageRepository();
+const notificationRepository = new NotificationRepository();
+const jobRepository = new JobRepository();
+const executionDetailsRepository = new ExecutionDetailsRepository();
+const organizationRepository = new OrganizationRepository();
+const environmentRepository = new EnvironmentRepository();
+const now = Date.now();
+const expireAtOneMonth = addMonths(now, 1);
+const expireAtSixMonths = addMonths(now, 6);
+
+export async function createExpireAt() {
+  // eslint-disable-next-line no-console
+  console.log('start migration - add expireAt field');
+
+  // eslint-disable-next-line no-console
+  console.log('get organizations and its environments');
+
+  const organizations = await organizationRepository.find({});
+  for (const organization of organizations) {
+    const environments = await environmentRepository.findOrganizationEnvironments(organization._id);
+    for (const environment of environments) {
+      const query = {
+        _organizationId: organization._id,
+        _environmentId: environment._id,
+        expireAt: { $exists: false },
+      };
+      await messagesSetExpireAt(query);
+      await notificationExpireAt(query);
+    }
+  }
+
+  // eslint-disable-next-line no-console
+  console.log('end migration');
+}
+
+export async function messagesSetExpireAt(query) {
+  await messageRepository.update(
+    {
+      ...query,
+      channel: { $ne: 'in_app' },
+    },
+    { $set: { expireAt: expireAtOneMonth } }
+  );
+
+  await messageRepository.update(
+    {
+      ...query,
+      channel: 'in_app',
+    },
+    { $set: { expireAt: expireAtSixMonths } }
+  );
+}
+
+export async function notificationExpireAt(query) {
+  const excludedIds = await getExcludedNotificationIds(query);
+
+  await notificationRepository.update(
+    { ...query, _id: { $nin: excludedIds } },
+    { $set: { expireAt: expireAtOneMonth } }
+  );
+  await jobRepository.update(
+    { ...query, _notificationId: { $nin: excludedIds } },
+    { $set: { expireAt: expireAtOneMonth } }
+  );
+  await executionDetailsRepository.update(
+    { ...query, _notificationId: { $nin: excludedIds } },
+    { $set: { expireAt: expireAtOneMonth } }
+  );
+}
```

Comment:
won't we override the entities which already have the `expireAt` set? Like we've deployed the changes with `expireAt` and now we want to run this migration as far as I understand.

Reply:
In query there is { expireAt: { $exists: false }}, so it will only be for those that do not have the field yet